app/src/main/java/co/nimblehq/compose/crypto/ui/screens/home/HomeViewModel.kt

```diff
@@ -103,6 +111,42 @@ class HomeViewModel @Inject constructor(
         _showTrendingCoinsLoading.value = false
     }
 
+    @OptIn(FlowPreview::class)
+    fun refresh() = execute {
```

Comment:
@Wadeewee I think let's keep it there, refreshing would need to be handled differently with the `init` state, since I supposed like this:

- Init: Send API request to get MyCoins and TrendingCoins and show on the UI once each receive the response
- Refresh: Wait for 2 API response complete and refresh 2 list at the same time

How do you think?

Reply:
> refreshing would need to be handled differently with the init state

@hoangnguyen92dn Normally, If we have the pull to refresh whenever users enter the screen, they should see `pull to refresh,` but for our current implementation, we don't show pull to refresh for first entering, so how does the user know we have to pull to refresh 🤔

Reply:
@Wadeewee It looks like weird behavior tbh 😄 The name itself shows the meaning, "refresh" not "initial" loading. Anw, I am still fine with your suggestion, how do you think? @luongvo

Reply:
@hoangnguyen92dn I made a draft code here https://github.com/nimblehq/jetpack-compose-crypto/commit/fddd854d2cd9123ab806b15b3fe85d0c7c6f4fd3, feel free to check it out 👋 

Notice that, there is 2 `delay` for demo only 🚧 

```
    init {
        loadData()
    }

    override fun loadData(isRefreshing: Boolean) {
        getMyCoins(isRefreshing)
        getTrendingCoins(isRefreshing)
    }

    private fun getMyCoins(isRefreshing: Boolean) = execute {
        _showMyCoinsLoading.value = true
        if (isRefreshing) showLoading()
        delay(2000)
        getMyCoinsUseCase.execute(
            GetMyCoinsUseCase.Input(
                currency = FIAT_CURRENCY,
                order = MY_COINS_ORDER,
                priceChangeInHour = MY_COINS_PRICE_CHANGE_IN_HOUR,
                itemPerPage = MY_COINS_ITEM_PER_PAGE,
                page = MY_COINS_INITIAL_PAGE
            )
        )
            .catch { e ->
                _error.emit(e)
            }
            .collect { coins ->
                _myCoins.emit(coins.map { it.toUiModel() })
            }
        _showMyCoinsLoading.value = false
        if (isRefreshing) hideLoading()
    }

    private fun getTrendingCoins(isRefreshing: Boolean) = execute {
        _showTrendingCoinsLoading.value = true
        if (isRefreshing) showLoading()
        delay(3000)
        getTrendingCoinsUseCase.execute(
            GetTrendingCoinsUseCase.Input(
                currency = FIAT_CURRENCY,
                order = MY_COINS_ORDER,
                priceChangeInHour = MY_COINS_PRICE_CHANGE_IN_HOUR,
                itemPerPage = MY_COINS_ITEM_PER_PAGE,
                page = MY_COINS_INITIAL_PAGE
            )
        )
            .catch { e ->
                _error.emit(e)
            }
            .collect { coins ->
                _trendingCoins.emit(coins.map { it.toUiModel() })
            }
        _showTrendingCoinsLoading.value = false
        if (isRefreshing) hideLoading()
    }
```


https://user-images.githubusercontent.com/16315358/198982652-4f96c8f5-f825-4aef-b44d-1700284111dc.mp4

Reply:
@luongvo As you see, while refreshing the data, the 2 coins list disappeared and showed the loading indicator with the refreshing indicator at the same time. With the UX, I believe it better to keep 2 lists there and only update once completed, if there is an error, the old data is still showing.

Reply:
> As you see, while refreshing the data, the 2 coins list disappeared and showed the loading indicator with the refreshing indicator at the same time.

@hoangnguyen92dn that could be fixed easily by adding a check into each data request block. The idea is keeping the consistent & having 1 `showLoading` to catch and wait for finishing all API requests which we already done before 🤔 

```
    init {
        loadData()
    }

    override fun loadData(isRefreshing: Boolean) {
        getMyCoins(isRefreshing)
        getTrendingCoins(isRefreshing)
    }

    private fun getMyCoins(isRefreshing: Boolean) = execute {
        if (isRefreshing) showLoading() else _showMyCoinsLoading.value = true
        delay(2000)
        getMyCoinsUseCase.execute(
            GetMyCoinsUseCase.Input(
                currency = FIAT_CURRENCY,
                order = MY_COINS_ORDER,
                priceChangeInHour = MY_COINS_PRICE_CHANGE_IN_HOUR,
                itemPerPage = MY_COINS_ITEM_PER_PAGE,
                page = MY_COINS_INITIAL_PAGE
            )
        )
            .catch { e ->
                _error.emit(e)
            }
            .collect { coins ->
                _myCoins.emit(coins.map { it.toUiModel() })
            }
        if (isRefreshing) hideLoading() else _showMyCoinsLoading.value = false
    }

    private fun getTrendingCoins(isRefreshing: Boolean) = execute {
        if (isRefreshing) showLoading() else _showTrendingCoinsLoading.value = true
        delay(3000)
        getTrendingCoinsUseCase.execute(
            GetTrendingCoinsUseCase.Input(
                currency = FIAT_CURRENCY,
                order = MY_COINS_ORDER,
                priceChangeInHour = MY_COINS_PRICE_CHANGE_IN_HOUR,
                itemPerPage = MY_COINS_ITEM_PER_PAGE,
                page = MY_COINS_INITIAL_PAGE
            )
        )
            .catch { e ->
                _error.emit(e)
            }
            .collect { coins ->
                _trendingCoins.emit(coins.map { it.toUiModel() })
            }
        if (isRefreshing) hideLoading() else _showTrendingCoinsLoading.value = false
    }
```

Reply:
@luongvo Deal 🤝 We will go with this approach.